thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/RootCauseMetricResource.java

```diff
@@ -356,11 +358,15 @@ public double getAggregate(@ApiParam(value = "metric urn", required = true) @Que
 
     data = data.dropNull();
 
+    StringSeries dimNames = data.getStrings(COL_DIMENSION_NAME);
     for (int i = 0; i < data.size(); i++) {
       final String dimName = data.getString(COL_DIMENSION_NAME, i);
       final String dimValue = data.getString(COL_DIMENSION_VALUE, i);
       final double value = data.getDouble(COL_VALUE, i);
-
+      // remove group by dimensions which also have topk
+      if (dimNames.contains(dimName + TOP_K_POSTFIX)) {
+        continue;
+      }
       if (!output.containsKey(dimName)) {
```

Comment:
can we use treemap to store output, so that the order of dimensions is alphabetical

Reply:
yes, I can add that.